core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -505,7 +507,14 @@ class KafkaServer(
             KafkaServer.MIN_INCREMENTAL_FETCH_SESSION_EVICTION_MS))
 
         // Start RemoteLogManager before broker start serving the requests.
-        remoteLogManager.foreach(_.startup())
+        remoteLogManagerOpt.foreach(rlm => {
+          val listenerName = ListenerName.normalised(config.remoteLogManagerConfig.remoteLogMetadataManagerListenerName())
+          val endpoint = brokerInfo.broker.endPoints.find(e => e.listenerName.equals(listenerName))
+            .getOrElse(throw new ConfigException(RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP +
+              " should be set as a listener name within valid broker listener name list."))
+          rlm.endPoint(Optional.of(endpoint))
```

Comment:
endpoint is guaranteed to be present (else we throw an error). We can remove the optional here.

Reply:
Good point. Updated.